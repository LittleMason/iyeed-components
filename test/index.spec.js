import VueLoadmore from '../packages/vuejs-loadmore/index';
import { mount } from '@vue/test-utils';
import { trigger } from './utils/event';

// 下拉测试
test('pull refresh', () => {
  const wrapper = mount(VueLoadmore, {
    propsData: {
      onRefresh: (done) => { done() },
    },
  });
  const track = wrapper.find('.vuejs-refresh-track');

  // pulling
  trigger(track, 'touchstart', 0, 0);
  trigger(track, 'touchmove', 0, 20);
  expect(wrapper.vm.$el).toMatchSnapshot();
  

  // loosing
  trigger(track, 'touchmove', 0, 75);
  trigger(track, 'touchmove', 0, 100);
  expect(wrapper.vm.$el).toMatchSnapshot();

  // loading
  trigger(track, 'touchend', 0, 100);
  expect(wrapper.vm.$el).toMatchSnapshot();

});