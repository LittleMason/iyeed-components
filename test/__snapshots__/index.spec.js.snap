// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull refresh 1`] = `
<div
  class="vuejs-loadmore-wrap"
>
  <div
    class="vuejs-refresh-track"
    style="transition-duration: 0ms; transform: translate3d(0, 20px, 0); webkit-transform: translate3d(0, 20px, 0);"
  >
    <div
      class="vuejs-refresh-head"
    >
      <div
        class="vuejs-refresh-text"
      >
        下拉刷新
      </div>
    </div>
      
    <div
      class="vuejs-loadmore"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`pull refresh 2`] = `
<div
  class="vuejs-loadmore-wrap"
>
  <div
    class="vuejs-refresh-track"
    style="transition-duration: 0ms; transform: translate3d(0, 75px, 0); webkit-transform: translate3d(0, 75px, 0);"
  >
    <div
      class="vuejs-refresh-head"
    >
      <div
        class="vuejs-refresh-text"
      >
        释放刷新
      </div>
    </div>
      
    <div
      class="vuejs-loadmore"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`pull refresh 3`] = `
<div
  class="vuejs-loadmore-wrap"
>
  <div
    class="vuejs-refresh-track"
    style="transition-duration: 200ms; transform: translate3d(0, 50px, 0); webkit-transform: translate3d(0, 50px, 0);"
  >
    <div
      class="vuejs-refresh-head"
    >
      <div>
        <div
          class="vuejs-loading vuejs-loading-circular"
        >
          <span
            class="vuejs-loading-spinner vuejs-loading-spinner-circular"
          >
            <svg
              class="vuejs-loading-circular"
              viewBox="25 25 50 50"
            >
              <circle
                cx="50"
                cy="50"
                fill="none"
                r="20"
              />
            </svg>
          </span>
           
          <span
            class="vuejs-loading-text"
          >
            正在刷新
          </span>
        </div>
      </div>
    </div>
      
    <div
      class="vuejs-loadmore"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
